src/libcollections/str.rs

```diff
@@ -263,22 +284,41 @@ impl str {
         core_str::StrExt::as_ptr(self)
     }
 
-    /// Takes a bytewise slice from a string.
+    /// Creates a string slice from another string slice, bypassing safety
+    /// checks.
+    ///
+    /// This new slice goes from `begin` to `end`, including `begin` but
+    /// excluding `end`.
     ///
-    /// Returns the substring from [`begin`..`end`).
+    /// To get a mutable string slice instead, see the
+    /// [`slice_mut_unchecked()`] method.
+    ///
+    /// [`slice_mut_unchecked()`]: #method.slice_mut_unchecked
     ///
     /// # Safety
     ///
-    /// Caller must check both UTF-8 sequence boundaries and the boundaries
-    /// of the entire slice as well.
+    /// Callers of this function are responsible that three sets of bounds are
+    /// satisifed:
+    ///
+    /// * `begin` must come before `end`.
+    /// * `begin` and `end` must be valid positions in the string slice.
+    /// * `begin` and `end` must lie on UTF-8 sequence boundaries.
```

Comment:
Ah, I see. No, I'm thinking #2 is like "both must be in the string" and #3 is like "any position in the string isn't valid, just sequence boundaries"

But I can see how reading #2 doesn't ge that across. Hm.